packages/groq-builder/src/commands/filterByType.ts

```diff
@@ -1,13 +1,16 @@
 import { GroqBuilder } from "../groq-builder";
-import { ResultItem, ResultOverride } from "../types/result-types";
+import { ResultItem } from "../types/result-types";
 import { ExtractTypeNames } from "../types/schema-types";
 
 declare module "../groq-builder" {
   export interface GroqBuilder<TResult, TRootConfig> {
-    filterByType<TType extends ExtractTypeNames<ResultItem<TResult>>>(
+    filterByType<TType extends ExtractTypeNames<ResultItem.Infer<TResult>>>(
       ...type: TType[]
     ): GroqBuilder<
-      ResultOverride<TResult, Extract<ResultItem<TResult>, { _type: TType }>>,
+      ResultItem.Override<
+        TResult,
+        Extract<ResultItem.Infer<TResult>, { _type: TType }>
+      >,
```

Comment:
If you want, you could probably import `Override` locally so you don't have to specify the `ResultItem.` prefix. I do this often in Rust, only using the namespace if there are conflicting symbols. So - if there's another `Override` type you're working with you'd use the fully-qualified form (`ResultItem.Override`) for both of the conflicting symbols, but otherwise you could just use `Override`.

I think it's as easy as `import Override = ResultItem.Override`, but you can decide whether it's worth doing or would be confusing.

Reply:
In this case,  I prefer the semantics of "Result Item Override" because that's more descriptive of what's happening.